android/src/main/java/com/reactnativeandroidwidget/builder/widget/utils/ResourceUtils.java

```diff
@@ -100,11 +129,24 @@ private static int getResourceId(Context context, String resourceType, String na
     }
 
     private static InputStream getInputStreamFromSource(String source) throws IOException {
-        URL url = new URL(source);
-        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
-        connection.setDoInput(true);
-        connection.connect();
-        return connection.getInputStream();
+        try {
+            URL url = new URL(source);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setDoInput(true);
+            connection.setConnectTimeout(15000);
+            connection.setReadTimeout(15000);
+            connection.connect();
+            
+            int responseCode = connection.getResponseCode();
+            if (responseCode != HttpURLConnection.HTTP_OK) {
+                connection.disconnect();
+                throw new IOException("HTTP error code: " + responseCode);
```

Comment:
Why throw `IOException` explicitly?
It will be thrown from `openConnection`, `connect, and `getInputStream`.
No need to catch a general exception, and throw it `IOException` again.

So no need for try/catch here, it should be handled from the caller place.


Maybe the only change in the PR can be in `ImageWidget.java` to check if the `bitmapFromURL` is null, and do an early return